### DISCUSSION TITLE
need help with python and pygame

### DISCUSSION THREAD

**Discussion by nivanda** [community/community#21520] [Other Feature Feedback, Questions, & Ideas]
<p>Hi I’m making a text based game in pygame with python 2.7</p>
<p>Here is the problem</p>
<pre><code>answer =input("Sa arkad ules pimedas ruumis, sa ei tea kuidas sa sinna sattusid. Mida sa teed? KOBA RINGI/OOTA")
if answer =="koba ringi":
answer =input("Sa kobad ringi ja leiad eseme mis tundub olevat valgumihkel. Kas sa tahad selle polema panna? JAH/EI")
if answer =="jah":
answer =input("See ei lahe polema. Peatselt sa kuuled samme, sa eeldad et sa oled kinnises ruumis. Mida sa teed? KARJU JUUST/OOTA")
if answer =="karju juust":
print "Sa saad millegiga pihta ja arkad ules oma toas voodis. See oli lihtsalt uni. SA VOITSID!"  
elif answer =="oota":
print "Keegi jooksis ruumi ja virutas sulle vastu pead. SA KAOTASID!"  
  
elif answer =="ei":
print "Keegi jooksis ruumi ja virutas sulle vastu pead. SA KAOTASID!"  
</code></pre>
<p> I want to move move between </p>
<pre><code>if answer =="karju juust":
</code></pre>
<p> and </p>
<pre><code>if answer =="koba ringi":
</code></pre>
<p>without ending the game</p>
<p>what is the best solution for that?</p>
<p>Sorry for my english and don’t mind the text (it is in my own language).</p>

**Comment by fire-eggs** [comment, marked as answer]
<p>Your sentence “I want to move between answer ‘a’ and answer ‘b’” is a little unclear, so I’m a little uncertain about what you’re trying to do.</p>
<p>If you are trying to re-use questions, or have additional sequences of questions, you might try using python functions to post questions and check answers. Since each of your questions are two-answer questions, you can return a boolean result from the functions.</p>
<pre><code>if (question1()):
  if (question2()):
    if (question3()):
      goodend()
    else:
      badend()
  else:
      badend()
else:
  if (question3()):
    goodend()
  else:
    badend()
</code></pre>
<p>So above, the question1() function asks your first question, returns true if the answer is “koba ringi”. question2() asks the second question, returns true if the answer is “jah”. question3() asks the third question, returns true if the answer is “karju juust”. But I’ve extended your logic so that if the answer to question1() is “oota”, then question3() is called: the player wins if the answer is “karju juust”.</p>
<p>Here is what the question1 function might look like:</p>
<pre><code>def question1():
  answer = input("Sa arkad ules pimedas ruumis, sa ei tea kuidas sa sinna sattusid. Mida sa teed? KOBA RINGI/OOTA")
  return answer == "koba ringi"
</code></pre>
<p>Asks the first question, and returns true if the answer is “koba ringi”.</p>
<p>I hope the functions “goodend” and “badend” are obvious: merely print the “you lived” / “you died” results.</p>
<p>This way, you can string together your questions in a different order, or re-use questions.</p>
<p>I hope this helps …</p>
<p>So to handle different possible answers without exiting, you’ll need a loop. When the loop exits, the game is over.</p>
<p>You’ll need to clarify what you want to do a little more. If you restated your sentence “I want to move between answer ‘a’ and answer ‘b’” to be a little clearer, we could help you more easily, and the answer may even occur to you on your own!</p>
